pkg/providers/common/common.go

```diff
@@ -97,3 +99,22 @@ func KubeadmConfigTemplateName(clusterName, workerNodeGroupName string, now type
 	t := now().UnixNano() / int64(time.Millisecond)
 	return fmt.Sprintf("%s-%s-template-%d", clusterName, workerNodeGroupName, t)
 }
+
+func IncrementName(name string) (string, error) {
+	re := regexp.MustCompile(`(-)(\d+)$`)
+	match := re.FindStringSubmatch(name)
+	if match == nil {
+		return "", fmt.Errorf(`invalid format of name [name=%s]. Name has to follow regex pattern "(-)(\d+)$", e.g. machinetemplate-cp-1`, name)
+	}
+
+	n, err := strconv.Atoi(match[2])
+	if err != nil {
+		return "", fmt.Errorf("converting object suffix to int: %v", err)
+	}
+
+	return re.ReplaceAllString(name, fmt.Sprintf("${1}%d", n+1)), nil
```

Comment:
can we have another method like
`ObjectName(baseName string, version int) string`

And we can reuse that from `DefaultObjectName`. That way there is only one place where the actual name is built.